content/cloud/variables.md

```diff
@@ -0,0 +1,191 @@
+
+In reality, you'd have multiple usernames and password hashes in a database, but for this tutorial we will configure the expected password using a Spin application variable. We'll name the variable `password` and set `required = true` since there is no reasonable default value. To do this, add a top-level `[variables]` section to the application manifest (`spin.toml`) and declare the variable within it.
+
+To surface the variable to the `pw-checker` component, add a `[component.config]` section in the component and specify the variable within it. Instead of statically assigning the value of the config variable, we are referencing the dynamic variable with [mustache](https://mustache.github.io/)-inspired string templates. The resulting application manifest should look similar to the following:
+
+> Note: only components that explicitly use the variables in their configuration section will get access to them. This enables only exposing variables and secrets to the desired components of an application.
+
+<!-- @selectiveCpy -->
+```toml
+spin_manifest_version = "1"
+description = "A Spin app with a dynamically updatable "
+name = "pw_checker"
+trigger = { type = "http", base = "/" }
+version = "0.1.0"
+
+[variables]
+password = { required = true }
+
+[[component]]
+id = "pw-checker"
+source = "app.wasm"
+[component.trigger]
+route = "/..."
+[component.build]
+command = "spin py2wasm app -o app.wasm"
+[component.config]
+password = "{{ password }}"
+```
+
+## Using Variables in a Spin Application
+
+Now that we have defined our variables and surfaced them to our component, we are ready to implement our Spin application. The application should get the user-provided password from the body of the HTTP request, compare it to the expected password set in our configuration variable, and authenticate accordingly. We will use the Spin `spin_config` module to retrieve the value of the `password` variable. 
+
+<!-- @selectiveCpy -->
+```py
+from spin_http import Response
+from spin_config import config_get
+
+def handle_request(request):
+    password = request.body.decode("utf-8")
+    expected = config_get("password")
+    access = "denied"
+    if expected == password:
+        access = "accepted"
+    response = f'{{"authentication": "{access}"}}'
+    return Response(200,
+                    {"content-type": "application/json"},
+                    bytes(response, "utf-8"))
+
+
+```
+
+Build and run the application locally to test it out. We will use the [environment variable provider](../spin/dynamic-configuration.md#environment-variable-provider) to set the variable values locally. The provider gets the variable values from the `spin` process's environment, searching for environment variables prefixed with `SPIN_CONFIG_`.
+
+<!-- @selectiveCpy -->
+```bash
+$ spin build
+$ SPIN_CONFIG_EXPECTED_PASSWORD="123" SPIN_CONFIG_PASSWORD="123" spin up
```

Comment:
Are you using `SPIN_CONFIG_EXPECTED_PASSWORD` in the code?

Reply:
great catch. didn't fully clean up when i removed it